File: DemoConstrain/app/src/main/java/com/example/democonstrain/MainActivity.java
package com.example.democonstrain;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    EditText edtA, edtB, edtKQ;
    Button btnTong;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main); // tên file XML của bạn

        // Ánh xạ view
        edtA = findViewById(R.id.edtA);
        edtB = findViewById(R.id.edtB);
        edtKQ = findViewById(R.id.edtKQ);
        btnTong = findViewById(R.id.btntong);

        // Xử lý sự kiện bấm nút
        btnTong.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                try {
                    // Lấy dữ liệu từ EditText
                    String strA = edtA.getText().toString().trim();
                    String strB = edtB.getText().toString().trim();

                    // Kiểm tra rỗng
                    if (strA.isEmpty() || strB.isEmpty()) {
                        Toast.makeText(MainActivity.this, "Vui lòng nhập đủ A và B", Toast.LENGTH_SHORT).show();
                        return;
                    }

                    // Chuyển sang số
                    double a = Double.parseDouble(strA);
                    double b = Double.parseDouble(strB);

                    // Tính tổng
                    double tong = a + b;

                    // Hiển thị kết quả
                    edtKQ.setText(String.valueOf(tong));

                } catch (NumberFormatException e) {
                    Toast.makeText(MainActivity.this, "Dữ liệu nhập không hợp lệ", Toast.LENGTH_SHORT).show();
                }
            }
        });
    }
}

File: ConvertLunarYear/app/src/main/java/com/example/convertlunaryear/MainActivity.kt
package com.example.convertlunaryear

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.example.convertlunaryear.R

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Bước 1: Ánh xạ View từ XML
        val editTextYear = findViewById<EditText>(R.id.editTextYear)
        val buttonConvert = findViewById<Button>(R.id.button1)
        val textViewResult = findViewById<TextView>(R.id.textViewResult)

        // Bảng Can & Chi
        val can = arrayOf("Giáp", "Ất", "Bính", "Đinh", "Mậu", "Kỷ", "Canh", "Tân", "Nhâm", "Quý")
        val chi = arrayOf("Tý", "Sửu", "Dần", "Mão", "Thìn", "Tỵ", "Ngọ", "Mùi", "Thân", "Dậu", "Tuất", "Hợi")

        // Bước 2: Gán sự kiện click cho nút
        buttonConvert.setOnClickListener {
            val yearInput = editTextYear.text.toString().trim()

            // Kiểm tra nhập trống
            if (yearInput.isEmpty()) {
                textViewResult.text = "Vui lòng nhập năm dương lịch!"
                return@setOnClickListener
            }

            // Kiểm tra nhập số hợp lệ
            val year = try {
                yearInput.toInt()
            } catch (e: NumberFormatException) {
                textViewResult.text = "Năm nhập không hợp lệ!"
                return@setOnClickListener
            }

            // Bước 3: Tính Can - Chi
            val canIndex = (year + 6) % 10
            val chiIndex = (year + 8) % 12
            val lunarYear = "${can[canIndex]} ${chi[chiIndex]}"

            // Hiển thị kết quả
            textViewResult.text = "Năm âm lịch: $lunarYear"
        }
    }
}
